test/checkChallenge.test.js

```diff
@@ -0,0 +1,1627 @@
+
+    // Just making sure it works similarly as byoyomi, so testing only "live" speed.
+
+    // Main time
+
+    it('reject main time for all the stones live too low', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "canadian", time_control: "canadian", speed: "live", stones_per_period: 5, main_time: 59, period_time: 1 } });
+  
+      config.minmaintimelive = 60;
+      config.maxmaintimelive = 1800;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'Minimum Main Time for live games in canadian is 1 minutes, please increase Main Time.' }));
+    });
+  
+    it('accept main time for all the stones live edge min', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "canadian", time_control: "canadian", speed: "live", stones_per_period: 5, main_time: 60, period_time: 80 } });
+  
+      config.minmaintimelive = 60;
+      config.maxmaintimelive = 1800;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, { reject: false });
+    });
+  
+    it('accept main time for all the stones live between min and max', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "canadian", time_control: "canadian", speed: "live", stones_per_period: 5, main_time: 120, period_time: 80 } });
+  
+      config.minmaintimelive = 60;
+      config.maxmaintimelive = 1800;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, { reject: false });
+    });
+  
+    it('accept main time for all the stones live edge max', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "canadian", time_control: "canadian", speed: "live", stones_per_period: 5, main_time: 1800, period_time: 80 } });
+  
+      config.minmaintimelive = 60;
+      config.maxmaintimelive = 1800;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, { reject: false });
+    });
+  
+    it('reject main time for all the stones live too high', () => {
+      const notification = base_challenge({ ranked: false, time_control: { system: "canadian", time_control: "canadian", speed: "live", stones_per_period: 5, main_time: 1801, period_time: 1800 } });
+  
+      config.minmaintimelive = 60;
+      config.maxmaintimelive = 1800;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'Maximum Main Time for live games in canadian is 30 minutes, please reduce Main Time.' }));
+    });
+
+    // Periods are not checked for non-byoyomi time controls
+
+    // Period Time (Period Time for all the X Stones)
+
+    it('reject period time for all the stones live too low', () => {
```

Comment:
oh
i thought canadian checks if for example 25 stones for 5 minutes means you get 5 minutes to play all the 25 stones, and if you play only 24 stones at the 5th minute end of time you lose by timeout no?

so it looks different from how the number of periods work, when if at the end of a periodtime of 5 minutes you didnt play your stone then you lose one number of periods, and you can still get a new periodtime and number of periods - 1 and continue to play as long as you have other periods

Reply:
Canadian only has 1 period in the sense byoyomi has periods. That one period only reset when you play the last stone required for that period. You could play 19 stones in 1 second and then wait 4:58 for the 20th stone, or you could play 1 stone in 4:50 and then do the other 20 in 9 seconds. However when you take 5 minutes for less than 20 stones, you lose on time.

Reply:
yes so it's as if in canadian there is only one period, and periodtime is `time per stone * number of stones`!

so putting a minimum or maximum number of periods in canadian does not make sense, because it is hardcoded in how canadian works (we could add --stonesperperiod`) or other options and make a separate getCanadianTimeSettingsRejectResult though